app/models/user.rb

```diff
@@ -55,6 +55,23 @@ def events
     accepted_applications.collect { |a| a.event }
   end
 
+  # Returns true iff. user has submitted an application letter for the given event
+  #
+  # @param [Event] given_event
+  # @return [Boolean]
+  def application_letter_for_event?(given_event)
+    return !self.application_letter_for_event(given_event).nil?
```

Comment:
I am not sure. [This style guide](https://github.com/airbnb/ruby#no-and-or) for example is strictly against the usage of not. The reason is that the `not` operation has a very low precedence whereas the `!` has one of the highest precedence just like I would expect from other languages. 

> !